cmd/lambda/populate_reset_queue/main.go

```diff
@@ -1,80 +1,85 @@
 package main
 
 import (
-	"encoding/json"
 	"log"
 
-	"github.com/pkg/errors"
-
-	"github.com/Optum/dce/pkg/common"
-	"github.com/Optum/dce/pkg/db"
+	"github.com/Optum/dce/pkg/account"
+	"github.com/Optum/dce/pkg/config"
+	"github.com/Optum/dce/pkg/event/eventiface"
 	"github.com/aws/aws-lambda-go/events"
 	"github.com/aws/aws-lambda-go/lambda"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/sqs"
 )
 
-// addAccountToQueue publishes a single account ID into the SQS
-// as an event for consumption
-func addAccountToQueue(accounts []*db.Account, queueURL *string,
-	queue common.Queue, dbSvc db.DBer) error {
-	// For each Account, send the message to Reset Queue and update
-	// FinanceLock Lease status if necessary
-	for _, acct := range accounts {
-		// Send Message
-		body, err := json.Marshal(acct)
-		if err != nil {
-			return errors.Wrapf(err, "Failed to add account %s to queue accounts", acct.ID)
-		}
-		sBody := string(body)
-		err = queue.SendMessage(queueURL, &sBody)
-		if err != nil {
-			return errors.Wrapf(err, "Failed to add account %s to queue accounts", acct.ID)
-		}
-		log.Printf("%s : Added to Reset Queue\n", acct.ID)
-	}
-	return nil
-
+type configuration struct {
+	Debug         string `env:"DEBUG" envDefault:"false"`
+	ResetQueueURL string `env:"RESET_SQS_URL" envDefault:"SqsUrl"`
 }
 
-// rbenqHandler is the base handler function for the lambda
-func rbenqHandler(cloudWatchEvent events.CloudWatchEvent) error {
+var (
+	services *config.ServiceBuilder
+	// Settings - the configuration settings for the controller
+	settings *configuration
+)
 
-	// Create Database Service
-	dbSvc, err := db.NewFromEnv()
-	if err != nil {
-		return err
+func init() {
+	cfgBldr := &config.ConfigurationBuilder{}
+	settings = &configuration{}
+	if err := cfgBldr.Unmarshal(settings); err != nil {
+		log.Fatalf("Could not load configuration: %s", err.Error())
 	}
 
-	// Get NotReady Accounts
-	accounts, err := dbSvc.FindAccountsByStatus(db.NotReady)
+	// load up the values into the various settings...
+	err := cfgBldr.WithEnv("AWS_CURRENT_REGION", "AWS_CURRENT_REGION", "us-east-1").Build()
```

Comment:
not necessarily a change for this PR, but this is still a little awkward to me how we're pulling in env vars like this, 
which are actually hidden requirements by our DB service.

It seems that if our DB service needs this env var, it's own service constructor should be in charge of declaring that.
I want our controller to only have to declare the env vars it's using directly.

Reply:
I agree with this.  Would be a nice fix.  No repetition if its always needed.